code/modules/mining/boulder_processing/boulder.dm

```diff
@@ -0,0 +1,279 @@
+	var/process_speed = 0
+	if(weapon)
+		process_speed = weapon.toolspeed
+		weapon.play_tool_sound(src, 50)
+		if(!continued)
+			to_chat(user, span_notice("You swing at \the [src]..."))
+	else if (override_speed_multiplier)
+		process_speed = override_speed_multiplier
+		playsound(src, 'sound/effects/rocktap1.ogg', 50)
+		if(!continued)
+			to_chat(user, span_notice("You scrape away at \the [src]..."))
+	else
+		return
+
+	if(!HAS_TRAIT(src, TRAIT_INSTANTLY_PROCESSES_BOULDERS))
+		if(!do_after(user, (2 * process_speed SECONDS), target = src))
+			return FALSE
+		if(!user.Adjacent(src))
+			return
+		durability--
+		user.apply_damage(4, STAMINA)
+	else
+		durability = 0
+	if(durability <= 0)
+		convert_to_ore()
+		to_chat(user, span_notice("You finish working on \the [src], and it crumbles into ore."))
+		playsound(src, 'sound/effects/rock_break.ogg', 50)
+		user.mind?.adjust_experience(/datum/skill/mining, MINING_SKILL_BOULDER_SIZE_XP * 0.2)
+		qdel(src)
+		return
+	var/msg = (durability == 1 ? "is crumbling!" : "looks weaker!")
+	to_chat(user, span_notice("\The [src] [msg]"))
+	manual_process(weapon, user, override_speed_multiplier, continued = TRUE)
+
+/**
+ * This function is called while breaking boulders manually, and drops ore based on the boulder's mineral content.
+ * Quantity of ore spawned here is 1 less than if the boulder was processed by a machine, but clamped at 10 maximum, 1 minimum.
+ */
+/obj/item/boulder/proc/convert_to_ore()
+	for(var/datum/material/picked in custom_materials)
+		var/obj/item/stack/ore/cracked_ore // Take the associated value and convert it into ore stacks...
+		var/quantity = clamp(round((custom_materials[picked] - SHEET_MATERIAL_AMOUNT)/SHEET_MATERIAL_AMOUNT), 1, 10) //but less resources than if they processed it by hand.
+		switch(picked.type)
+			if(/datum/material/iron)
+				cracked_ore = new /obj/item/stack/ore/iron(drop_location(), quantity)
+			if(/datum/material/gold)
+				cracked_ore = new /obj/item/stack/ore/gold(drop_location(), quantity)
+			if(/datum/material/silver)
+				cracked_ore = new /obj/item/stack/ore/silver(drop_location(), quantity)
+			if(/datum/material/plasma)
+				cracked_ore = new /obj/item/stack/ore/plasma(drop_location(), quantity)
+			if(/datum/material/diamond)
+				cracked_ore = new /obj/item/stack/ore/diamond(drop_location(), quantity)
+			if(/datum/material/glass)
+				cracked_ore = new /obj/item/stack/ore/glass/basalt(drop_location(), quantity)
+			if(/datum/material/bluespace)
+				cracked_ore = new /obj/item/stack/ore/bluespace_crystal(drop_location(), quantity)
+			if(/datum/material/titanium)
+				cracked_ore = new /obj/item/stack/ore/titanium(drop_location(), quantity)
+			if(/datum/material/uranium)
+				cracked_ore = new /obj/item/stack/ore/uranium(drop_location(), quantity)
+		SSblackbox.record_feedback("tally", "ore_mined", quantity, cracked_ore)
+
+/obj/item/boulder/proc/can_get_processed()
```

Comment:
Auto doc. or maybe you can move this one line to the place where it's needed cause you know it's just one line.